iothub/device/src/Authentication/IotHubConnectionInfo.cs

```diff
@@ -97,58 +99,50 @@ public IotHubConnectionInfo(IotHubConnectionStringBuilder builder)
             {
                 SharedAccessSignature = builder.SharedAccessSignature;
             }
+
+            AuthenticationModel = SharedAccessKeyName == null
+                ? AuthenticationModel.SasIndividual
+                : AuthenticationModel.SasGrouped;
         }
 
-        // This constructor is only used for unit testing.
-        internal IotHubConnectionInfo(
-            string ioTHubName = null,
-            string deviceId = null,
-            string moduleId = null,
-            string hostName = null,
-            Uri httpsEndpoint = null,
-            Uri amqpEndpoint = null,
-            string audience = null,
-            string sharedAccessKeyName = null,
-            string sharedAccessKey = null,
-            string sharedAccessSignature = null,
-            bool isUsingGateway = false)
+        // This contructor is intended for creating an IotHubConnectionInfo that uses an IAuthenticationMethod-based authentication.
+        internal IotHubConnectionInfo()
         {
-            IotHubName = ioTHubName;
-            DeviceId = deviceId;
-            ModuleId = moduleId;
-            HostName = hostName;
-            HttpsEndpoint = httpsEndpoint;
-            AmqpEndpoint = amqpEndpoint;
-            Audience = audience;
-            SharedAccessKeyName = sharedAccessKeyName;
-            SharedAccessKey = sharedAccessKey;
-            SharedAccessSignature = sharedAccessSignature;
-            IsUsingGateway = isUsingGateway;
+            // if certificate information is present, set:
+            // AuthenticationModel = AuthenticationModel.X509;
```

Comment:
What's the plans here?

Reply:
My plan is to use this constructor to create the IotHubConnectionInfo using an IAuthenticationMethod. Within this constructor we'll check if the certificate information is present, and set the AuthenticationModel in there.

Reply:
So, this is a TODO for an upcoming change?

Reply:
yes.